src/ServerRequestFactory.php

```diff
@@ -132,29 +153,47 @@ private function getUri(array $server): UriInterface
         return $uri;
     }
 
-    /**
-     * @suppress PhanUndeclaredFunction
-     */
     private function getHeadersFromGlobals(): array
     {
-        if (\function_exists('\getallheaders')) {
-            $headers = \getallheaders();
-            if ($headers === false) {
-                $headers = [];
+        if (function_exists('getallheaders')) {
+            $headers = getallheaders();
+            return $headers === false ? [] : array_filter($headers, static fn (string $value): bool => $value !== '');
+        }
+
+        $headers = [];
+
+        foreach ($_SERVER as $name => $value) {
+            if ($value === '') {
+                continue;
             }
-        } else {
-            $headers = [];
-            foreach ($_SERVER as $name => $value) {
-                if (strncmp($name, 'HTTP_', 5) === 0) {
-                    $name = str_replace(' ', '-', strtolower(str_replace('_', ' ', substr($name, 5))));
-                    $headers[$name] = $value;
+
+            if (strncmp($name, 'REDIRECT_', 9) === 0) {
```

Comment:
I am not sure if there is a place for `REDIRECT_` parameters among the headers. What parameter names do you mean?

Reply:
Apache prefixes environment variables with `REDIRECT_` if they are added by rewrite rules. :)

Reply:
These `REDIRECT_` variables are not headers.
I was checked two servers and wath i see:

`REDIRECT_STATUS = 200` (nginx)

and

```
REDIRECT_TZ = Europe/Moscow
REDIRECT_STATUS = 200
```
(nginx + apache)

Are you suggesting converting this to headers? Do not do it this way!

Reply:
This will not be converted to headers. The following parameters will be converted to headers: `REDIRECT_HTTP_*`, etc. Look carefully at the code!

Reply:
Can you write tests for this case?

Reply:
Yes